### ISSUE TITLE
Premium blocks: Prompt appears and disappears on focus when it should not appear at all.

### ISSUE THREAD

**Issue by kwight** [Automattic/jetpack#16997]
I found this while testing the upcoming Donations block. The prompt is for an "upgrade" to Personal, when the site is already on a Premium plan.

![2020-08-26 19 10 16](https://user-images.githubusercontent.com/349751/91375683-349ade80-e7d0-11ea-8d46-cf33786921ec.gif)

To replicate:
* Sandbox the API and the store.
* Create a new simple site with a Premium upgrade.
* Insert a Donations block.
* Click on the upgrade prompt.

**Comment by kwight** [comment]
@obenland is this Ajax territory?..

**Comment by obenland** [comment]
I can reproduce it with API and store sandboxed, but it seems to work fine in production? Any thoughts as to why sandboxing the store would change it?

**Comment by obenland** [comment]
Yeah looks like this is expected with Store sandboxed as it doesn't have access to production DBs.
See PCYsg-IA-p2#sandbox

Thanks for reporting though! I rather learn something about store sandboxing than leaving potential bugs in production.

**Comment by kwight** [comment]
> Yeah looks like this is expected with Store sandboxed as it doesn't have access to production DBs

Hm, what is it about this feature that requires the production dbs? The sandboxed store works the same way, and has plans just the same (for instance, previous prompts to upgrade worked the same regardless of sandboxing or not).

**Comment by obenland** [comment]
> Hm, what is it about this feature that requires the production dbs?

It needs the right plan associated with the site. My personal blog has a Premium plan but shows up as a free site with the store sandboxed.

**Comment by kwight** [comment]
Oh, I meant even when you buy the plan while sandboxed. I checked with the other Earn blocks, and it's only happening with the Donation block, so we'll take it from here.

**Comment by blackjackkent** [comment]
Copying some comments over from a PR I started:

I've started looking into this and am a little perplexed on where to find the logic that determines when this message is displayed at all.

The message itself appears to come from https://github.com/Automattic/jetpack/blob/master/extensions/extended-blocks/paid-blocks/upgrade-plan-banner.jsx which is then loaded into an HOC in https://github.com/Automattic/jetpack/blob/master/extensions/extended-blocks/paid-blocks/with-upgrade-banner.jsx. Less certain about where that's being used, though; it doesn't seem to be referenced directly in the donations block code so I'm assuming there's some middleware that is loading that behavior on top of any block that is marked with the plan_check property when it's registered.

So I'm still working on trying to figure out how that logic plugs into the initialization of the donations block and why it thinks the user doesn't have a premium plan for that block only.

**Comment by blackjackkent** [comment]
Down a bit of a rabbit hole on this one. 🤔 

I *believe* the reason the upgrade notification is being displayed is because it checks the value of a global window variable called `Jetpack_Editor_Initial_State` which is fetched in https://github.com/Automattic/jetpack/blob/master/extensions/shared/get-jetpack-extension-availability.js via a method in https://github.com/Automattic/jetpack/blob/master/extensions/shared/get-jetpack-data.js

I created a simple site with a premium plan as described in the issue and have verified that when I check this global variable in the console, it claims that the `donations` block type is unavailable due to the reason `"missing_plan"`.

![image](https://user-images.githubusercontent.com/13437011/92032769-429cb200-ed30-11ea-90e6-d7f7754103ad.png)

The only place I can find where this variable would be initialized is in https://github.com/Automattic/jetpack/blob/master/class.jetpack-gutenberg.php, where the `enqueue_block_editor_assets()` method includes the following:

```php
        wp_localize_script(
		'jetpack-blocks-editor',
		'Jetpack_Editor_Initial_State',
		array(
			'available_blocks' => self::get_availability(),
			'jetpack'          => array(
				'is_active'                 => Jetpack::is_active(),
				'is_current_user_connected' => $is_current_user_connected,
				/** This filter is documented in class.jetpack-gutenberg.php */
				'enable_upgrade_nudge'      => apply_filters( 'jetpack_block_editor_enable_upgrade_nudge', false ),
			),
			'siteFragment'     => $site_fragment,
			'adminUrl'         => esc_url( admin_url() ),
			'tracksUserData'   => $user_data,
			'wpcomBlogId'      => $blog_id,
			'allowedMimeTypes' => wp_get_mime_types(),
		)
	);
```

However, when I log out `l(self::get_availability())` at that point, it shows the donation block to be correctly available as expected.

<img width="1270" alt="Screen Shot 2020-09-02 at 3 38 38 PM" src="https://user-images.githubusercontent.com/13437011/92034508-f1da8880-ed32-11ea-98dc-a45ef14f262a.png">

I'm not sure where else that `Jetpack_Editor_Initial_State` value might be being set; @mmtr / @kwight (or anyone else), do you have any insight on this?

**Comment by kraftbj** [comment]
I answered in Slack, but on `jetpack_register_block`, there is a function to check `jetpack_register_gutenberg_extensions`, which is adds an action to the `jetpack_register_gutenberg_extensions` hook. My initial thought in running `l(self:get_availability())` that the hook hadn't fired so isn't populating that as expected.

**Comment by kwight** [comment]
Thanks @kraftbj ! Now, neither myself nor @blackjackkent can replicate, so we'll consider it a transient thing ¯\_(ツ)_/¯

**Comment by kwight** [comment]
Came across this again today: totally user error. It was because the store was sandboxed, but the site in question was not 🙃
